lib/inspector.js

```diff
@@ -131,6 +133,9 @@ class Session extends EventEmitter {
 }
 
 function inspectorOpen(port, host, wait) {
+  if (isEnabled()) {
+    throw new ERR_INSPECTOR_ALREADY_ACTIVATED();
```

Comment:
With this change, there's no way to wait for a debugger after opening it. We use this pattern in the VS Code debugger to open the inspector, pass its URL to our debugging logic, and then wait for the debugger to attach.

Can we only throw if `port !== undefined || host !== undefined`?

Reply:
IIUC the pattern goes like this:

```js
const openedFromCli = inspector.url() !== undefined;
if (!openedFromCli) {
  inspector.open(0, undefined, false); // first call to set the inspector.url()
}
inspector.open(openedFromCli ? undefined : 0, undefined, true);
```

Does changing the last call of `inspector.open()` to `inspector.waitForDebugger()` works? Previously, calling it the second time with `inspector.open(undefined, undefined, true)` roughly did the same thing as `inspector.waitForDebugger()` other than that it resets the port integer and the host string stored on the native side without actually doing anything about them (and that was why the url returned again showed these modified values).

Reply:
ah, neat! I didn't know we got that method, that's much cleaner. I'll use that method when available.